addons/mail/static/tests/translation/translation_test.js

```diff
@@ -0,0 +1,39 @@
+/* @odoo-module */
+
+import { click, start, startServer, contains } from "@mail/../tests/helpers/test_utils";
+
+QUnit.module("Google Cloud Translation");
+
+QUnit.test("Toggle display of original/translated version of chatter message", async (assert) => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({});
+    pyEnv["mail.message"].create({
+        model: "res.partner",
+        body: "Al mal tiempo, buena cara.",
+        author_id: pyEnv.odoobotId,
+        res_id: partnerId,
+    });
+    const { openFormView } = await start({
+        mockRPC(route) {
+            if (route === "/discuss/message/translate") {
+                assert.step("Request");
+                return { value: "To bad weather, good face.", source: "Spanish", error: null };
+            }
+        },
+    });
+    await openFormView("res.partner", partnerId);
+    await contains("button[aria-label='Translate']");
+    await contains("button[aria-label='Revert']", { count: 0 });
+    // Click acts as a toogle affecting its appearence and the actual message content displayed.
+    await click("button[aria-label='Translate']");
+    await contains(
+        `.o-mail-Message:contains("To bad weather, good face. (Translated from: Spanish)")`
+    );
+    await contains("button[aria-label='Translate']", { count: 0 });
+    await contains("button[aria-label='Revert']");
+    await click("button[aria-label='Revert']");
+    await contains(`.o-mail-Message:contains("Al mal tiempo, buena cara.")`);
+    await click("button[aria-label='Translate']");
+    // The translation button should not trigger more than one external request for a single message.
+    assert.verifySteps(["Request"]);
```

Comment:
It's better to assert something in the interface than technical line in the mock.

Reply:
There was misunderstanding, this test exists to ensure we do not trigger multiple rpc requests if the translation value is still somewhere in memory. Hence, reducing communication costs and improving UX.